.github/workflows/docs_preview.yml

```diff
@@ -0,0 +1,44 @@
+name: Deploy Docs Preview
+
+on:
+  pull_request_target:
+    paths:
+      - "mkdocs.yml"
+      - "docs/**"
+
+permissions:
+  contents: read
+  pull-requests: write
+  deployments: write
+  statuses: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: "deploy-to-netlify"
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
AIUI, on a pull_request_target trigger this checks out the repository on the target branch and not the pull request (quite intentionally), so this won't do what you want.

Reply:
ah, okay. The docs around this use "in the context" which is a thing I don't know what it is and I couldn't tell which checkout you get.

Reply:
https://github.com/orgs/community/discussions/22363#discussioncomment-3236409 is probably a better summary than the docs themselves (oops)